python/lsst/afw/coord/refraction.py

```diff
@@ -0,0 +1,284 @@
+
+    Parameters
+    ----------
+    wavelength : float
+        wavelength is in nm (valid for 230.2 < wavelength < 2058.6)
+    elevation : lsst.afw.geom Angle
+        Elevation of the observation, as an Angle.
+    weather : lsst.afw.coord Weather, optional
+        Class containing the measured temperature, pressure, and humidity
+        at the observatory during an observation
+    observatory : lsst.afw.coord Observatory, optional
+        Class containing the longitude, latitude, and altitude of the observatory.
+
+    Returns
+    -------
+    lsst.afw.geom Angle
+        The angular refraction for light of the given wavelength, under the given observing conditions.
+    """
+    latitude = observatory.getLatitude()
+    altitude = observatory.getElevation()
+
+    reducedN = deltaN(wavelength, weather)*1E-8
+
+    temperature = _extractTemperature(weather, useKelvin=True)
+    atmosScaleheightRatio = float(4.5908E-6*temperature/u.Kelvin)
+
+    # Account for oblate Earth
+    relativeGravity = (1. + 0.005302*np.sin(latitude.asRadians())**2. -
+                       0.00000583*np.sin(2.*latitude.asRadians())**2. - 0.000000315*altitude)
+
+    # Calculate the tangent of the zenith angle.
+    tanZ = np.tan(np.pi/2. - elevation.asRadians())
+
+    atmosTerm1 = reducedN*relativeGravity*(1. - atmosScaleheightRatio)
+    atmosTerm2 = reducedN*relativeGravity*(atmosScaleheightRatio - reducedN/2.)
+    result = Angle(float(atmosTerm1*tanZ + atmosTerm2*tanZ**3.))
+    return result
+
+
+def differentialRefraction(wavelength, wavelengthRef, elevation,
+                           weather=lsstWeather, observatory=lsstObservatory):
+    """Calculate the differential refraction between two wavelengths.
+
+    Parameters
+    ----------
+    wavelength : float
+        wavelength is in nm (valid for 230.2 < wavelength < 2058.6)
+    wavelengthRef : float
+        Reference wavelength, typically the effective wavelength of a filter.
+    elevation : lsst.afw.geom Angle
+        Elevation of the observation, as an Angle.
+    weather : lsst.afw.coord Weather, optional
+        Class containing the measured temperature, pressure, and humidity
+        at the observatory during an observation
+    observatory : lsst.afw.coord Observatory, optional
+        Class containing the longitude, latitude, and altitude of the observatory.
+
+    Returns
+    -------
+    lsst.afw.geom Angle
+        The refraction at `wavelength` - the refraction at `wavelengthRef`.
+    """
+    refractionStart = refraction(wavelength, elevation, weather=weather, observatory=observatory)
+    refractionEnd = refraction(wavelengthRef, elevation, weather=weather, observatory=observatory)
+    return refractionStart - refractionEnd
+
+
+def deltaN(wavelength, weather):
+    """Calculate the differential refractive index of air.
+
+    The differential refractive index is the difference of the refractive index from 1.,
```

Comment:
Too long for a docstring line: limit is 80 chars (though I can't find the specific item in the dev guide).